corehq/messaging/smsbackends/whatsapp/models.py

```diff
@@ -0,0 +1,61 @@
+from __future__ import absolute_import
+from collections import namedtuple
+
+from yowsup.demos.sendclient import YowsupSendStack
+from yowsup.layers import YowLayerEvent
+from yowsup.layers.auth import AuthError
+from yowsup.layers.network import YowNetworkLayer
+
+from corehq.apps.sms.models import SQLSMSBackend
+from corehq.apps.sms.util import clean_phone_number
+from corehq.messaging.smsbackends.whatsapp.forms import WhatsAppBackendForm
+from dimagi.utils import logging
+
+
+YowsupMessage = namedtuple('YowsupMessage', 'phone body')
+
+
+class SQLWhatsAppBackend(SQLSMSBackend):
```

Comment:
does the integration do any automatic rate limiting? if not a good rule of thumb to start off with would be to limit to 60 SMS per minute for the long code. to do that you can just add a method here, details [here](http://commcare-hq.readthedocs.io/messaging/sms_backends.html#rate-limiting).

Reply:
Thanks!